core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -239,31 +229,27 @@ protected function prepare($objPage)
 	 */
 	protected function getPageLayout($objPage)
 	{
-		$blnMobile = ($objPage->mobileLayout && Environment::get('agent')->mobile);
+		$objLayout = LayoutModel::findByPk($objPage->layout);
 
-		// Override the autodetected value
-		if (Input::cookie('TL_VIEW') == 'mobile')
-		{
-			$blnMobile = true;
-		}
-		elseif (Input::cookie('TL_VIEW') == 'desktop')
+		// HOOK: modify the page or layout object (see #4736)
+		if (isset($GLOBALS['TL_HOOKS']['getPageLayout']) && \is_array($GLOBALS['TL_HOOKS']['getPageLayout']))
```

Comment:
Why did you move the hook?

Reply:
Because the method says `getPageLayout()`. So I figured that's the correct way to place the hook. The order is still the same, it finds the assigned layout and then calls the hook.